frontend/src/i18n/locales/keys.test.ts

```diff
@@ -0,0 +1,107 @@
+import { error } from 'console';
+import fs from 'fs';
+import glob from 'glob';
+
+const allowlist = require('./allowlist.json');
+
+/*
+ * Description:
+ * This test will check for duplicate keys in the translation files and will fail if any are found.
+ * The allowlist.json file is used to ignore any keys that are intentionally duplicated.
+ * - The test will gather all the keys from the translation files and store them in a master list
+ * - Then it will organize the master list by language in a wordMap object
+ * - Finally it will check for duplicates in each language and fail if any are found
+ */
+async function checkKeys() {
+  const files = glob.sync('./**/i18n/locales/*/*.json', { ignore: '**/node_modules/**' });
+
+  /*
+   * Allowlist function scans the json and return if the word is within the allowlist
+   */
+  function allowlistScan(word: string, lang: string) {
+    return allowlist[lang].some((item: any) => item.wordKey === word);
+  }
+
+  /*
+   * The contents array is a master list meant to hold every word object before sorting
+   */
+  const contents: any[] = [];
+
+  for (const file of files) {
+    const content = await fs.readFileSync(file, 'utf-8');
+    const words = JSON.parse(content);
+
+    const filePart = file.split('/');
+    const language = filePart[filePart.length - 2];
+    const fileName = filePart[filePart.length - 1];
```

Comment:
I think we should try to use:
https://nodejs.org/api/path.html#pathdirnamepath
https://nodejs.org/api/path.html#pathbasenamepath-suffix